### PR TITLE
MINIFICPP-582: Add constraints for gcc in linux

### PR DESCRIPTION
**Pull request by phrocker**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: linux.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#!/bin/bash
+verify_gcc_enable(){
+  feature="$1"
+  feature_status=${!1}
+  if [ "$feature" = "BUSTACHE_ENABLED" ]; then
```
Comment (apiri) [inline linux.sh]: Should we also perform a check for Lua given its constraints in the README?

Comment (phrocker) [reply linux.sh]: @apiri looks like there was a minor collision with another PR that fixed an issue addressed here so I'll go ahead and resolve that conflict and implement the LUA constraints. Thanks!

File: linux.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#!/bin/bash
+verify_gcc_enable(){
+  feature="$1"
+  feature_status=${!1}
+  if [ "$feature" = "BUSTACHE_ENABLED" ]; then 
+    if (( COMPILER_MAJOR == 6 && COMPILER_MINOR >= 3 && COMPILER_REVISION >= 1  )); then
```
Comment (phrocker) [inline linux.sh]: @apiri updated. Broke the logic out here to make it cleaner ( or so I think ), and I was also missing some revision info for the bustache feature.

Comment (apiri) [reply linux.sh]: these changes look good, thanks for updating.  will merge in.

**Pull request closed without merging**